File: vol1/ch5/LeetCode350.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        m,n=len(nums1),len(nums2)
        if m>n: return self.intersect(nums2,nums1)
        hmap={}
        for x in nums1:
            if x in hmap:hmap[x]+=1
            else: hmap[x]=1
        ans=[]                    #存放答案
        for y in nums2:
            if y in hmap and hmap[y]>0:
                ans.append(y)
                hmap[y]-=1
        return ans

File: vol1/ch5/LeetCode350.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        int m=nums1.size();
        int n=nums2.size();
        if (m>n) return intersect(nums2,nums1);
        unordered_map <int,int> hmap;
        for (int x:nums1) hmap[x]++;
        vector<int> ans;                    //´æ·Å´ğ°¸
        for (int y:nums2) {
            if (hmap[y]>0) {
                ans.push_back(y);
                --hmap[y];
            }
        }
        return ans;
    }
};

